internal/builders/generic/README.md

```diff
@@ -1,6 +1,16 @@
 # Generation of SLSA3+ provenance for arbitrary projects
 
-This document explains how to use the builder for projects for which there is no language-specific builder available.
+This document explains how to generate SLSA provenance for projects for which
+there is no language or ecosystem specific builder available.
+
+This can be done by adding an additional step to your existing Github Actions
+workflow to call a [Github Actions reusable
+workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows).
+We'll call this the "generic workflow" from now on.
+
+The generic workflow differs from ecosystem specific builders (like the [Go
+builder](../go)) which build the artifacts as well as generate provenance. This
+project simply generates provenance as a separate step in an existing workflow.
```

Comment:
Mention something about the value this adds:

```Using this will generate a non-forgeable attestation to the artifacts' digests using the identity of the GitHub workflow. This can be used to create a positive attestation to a software artifact coming from your repository.```

Reply:
I added a section on benefits. PTAL